hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/handlers/MarkerHandler.java

```diff
@@ -150,7 +168,50 @@ public boolean doesMarkerDirExist(String markerDir) {
    * @param markerName marker name
    * @return the {@code CompletableFuture} instance for the request
    */
-  public CompletableFuture<String> createMarker(Context context, String markerDir, String markerName) {
+  public CompletableFuture<String> createMarker(Context context, String markerDir, String markerName,
+                                                String batchInterval, String period, String maxAllowableHeartbeatIntervalInMs,
+                                                String basePath, String earlyConflictDetectionEnable,
+                                                String earlyConflictDetectionClassName) {
+    // Step1 do early conflict detection if enable
```

Comment:
Should the async conflict detection thread be started in the constructor of the `MarkerHandler`?  It does not make sense to start the checker for every request.  Anyway, the checker is doing the detection in a batching and async way.

Reply:
Yeap, for now we init detection thread in the first time call this create marker API, and only fresh this thread when the instant time changed. 

We can also init this thread in constructor, but for now, we don't pass hoodieWriteConfig in MarkerHandler..